src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -485,11 +494,16 @@ async def _stop(self) -> None:
 _DATA_PIPELINE: _DataPipeline | None = None
 
 
-async def initialize(resampler_config: ResamplerConfig) -> None:
+async def initialize(
+    resampler_config: ResamplerConfig,
+    api_power_request_timeout: timedelta = timedelta(seconds=5.0),
+) -> None:
     """Initialize a `DataPipeline` instance.
 
     Args:
         resampler_config: Config to pass on to the resampler.
+        api_power_request_timeout: Timeout to use when making power requests to
+            the microgrid API.
```

Comment:
Maybe it would be good to document what happens if it times out? As a user I'm not sure what to do with this. Why should I increase or decrease the timeout?

Reply:
I've added a few more lines.  The only thing is that I've not been specific about the exact duration for which the components get blocked, because currently it is hard-coded to exponentially increase from 1s to a max of 30s when there are subsequent timeouts.

I think we don't have to provide a way to change the duration until we can do so from the UI.